net/devif/devif_poll.c

```diff
@@ -617,12 +617,89 @@ static inline int devif_poll_tcp_connections(FAR struct net_driver_s *dev,
 # define devif_poll_tcp_connections(dev, callback) (0)
 #endif
 
+/****************************************************************************
+ * Name: devif_poll_callback
+ *
+ * Description:
+ *   This function will help us to gather multiple iob memory slices into a
+ *   linear device buffer. if devices with small memory, this function will
+ *   trigger a memory copy if net device start transmit the iob slices to
+ *   flat buffer
+ *
+ ****************************************************************************/
+
+static int devif_poll_callback(FAR struct net_driver_s *dev)
+{
+  FAR struct iob_s *iob;
+  uint16_t llhdrlen;
+  int bstop;
+
+  if (dev->d_len == 0 || dev->d_iob == NULL)
+    {
+      return 0;
+    }
+
+  /* Loopback messge ? */
+
+  bstop = devif_loopback(dev);
+  if (bstop)
+    {
+      return bstop;
+    }
+
+  llhdrlen = NET_LL_HDRLEN(dev);
+
+  /* Copy iob to flat buffer */
+
+  iob_copyout(dev->d_pollbuf + llhdrlen, dev->d_iob, dev->d_len, 0);
```

Comment:
let's move the following code to devif_poll to avoid add the field d_pollcallback and d_pollbuf

Reply:
I think you should take a closer look at the implementation here. devif_poll_callback is called from l3/l4 stack, In the l3/l4 stack, we cannot guarantee that this pointer will not be changed